### DISCUSSION TITLE
MQTT Multiple Topics (US states)?

### DISCUSSION THREAD

**Discussion by chairmanfaust** [meshtastic/Discussions#107] [General]
My mqtt server currently bridges to the meshtastic public mqtt, from the mosquitto.conf:

```
topic msh/US/# out 0
topic msh/US/# in 0
````

My radio points to my server and is configured with the root topic msh/US/PA. This works and allows me to see other users in the state. Is it possible to configure either of these settings so that i can get surrounding states like NJ, NY, MD, etc? 

Here is the rest of my config:
```
per_listener_settings true
pid_file /run/mosquitto/mosquitto.pid
persistence true
persistence_location /var/lib/mosquitto/
log_dest file /var/log/mosquitto/mosquitto.log
include_dir /etc/mosquitto/conf.d
allow_anonymous false
listener 1883
password_file /etc/mosquitto/passwd

connection bridge-01
address mqtt.meshtastic.org:1883
try_private true

remote_username meshdev
remote_password large4cats
max_queued_messages 0

topic msh/US/# out 0
topic msh/US/# in 0
```

**Comment by garthvh** [comment]
Only one topic is allowed in the mqtt config, and the mqtt config is at the size limit possible to be used for admin messages.

**Reply by chairmanfaust** [reply]
is there a better or different way i should configure this, or is this pretty much where things are at for the moment with creating regional topics?

**Reply by garthvh** [reply]
The point of the topics is to better localize your device or phone with less traffic, not join together two large topics.

**Reply by chairmanfaust** [reply]
I get that, problem is i am closer to 3 other states than the rest of the one I'm in. is there a way to see what other topics exist? I'm curious if there are any besides just states that get traffic. A metropolitan regional topic would serve me better than a state one.

**Reply by garthvh** [reply]
You can make whatever topic you want, getting others to use it is the challenge.

**Reply by chairmanfaust** [reply]
thanks for the advice... i came across https://mqtt-explorer.com/ as a way to visualize the live topics in use, seems like a good place to start for zeroing in on more specific topics already in use... but as you said, you have to get people onboard for more regional topics... i do wish there was a way to subscribe to more topics... if there was a per channel way to at least

<img width="220" alt="image" src="https://github.com/user-attachments/assets/bf3fc3b3-a7a5-4db8-b527-5ab2f4f050fb">

**Reply by TerryFrench** [reply]
If you create your own MQTT server you could subscribe it (or some internet client) to the 3 topics you are after and publish that to one single topic on your server. 
You would at least discover the 3 states nodes on your device. 
I suppose that reflecting back messages day from PA to NY would be frowned upon by NY users. 
So for your node to upload to 3 places you could configure your MQTT topic so that when your nodes post a new message, you would proxy (send a copy of it) it to the meshtastic's server to the 3 topics you are after. 

That should enable you to see 3 states and nodes in 3 states to see your node and communicate with it. That without spamming or flooding the Meshtastic server. 

If you are not into scripting, mosquito and NodeRed might be a visual way to process messages across your server and the official one.

Note: i have not tried it.
